OptionSelector/CategoryTableViewController.m

```diff
@@ -0,0 +1,118 @@
+//
+//  CategoryTableViewController.m
+//  OptionSelector
+//
+//  Created by Henna on 8/10/15.
+//  Copyright (c) 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "CategoryTableViewController.h"
+#import "CQCategory.h"
+#import "OptionsTableViewController.h"
+#import "CategoryModel.h"
+
+
+@interface CategoryTableViewController ()
+@property (nonatomic) CategoryModel * data;
+
+@end
+
+@implementation CategoryTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.title = @"Henna's Categories";
+    self.data = [CategoryModel sharedInstance];
+    [self.data initializeModel];
+    
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+[self.tableView reloadData];
```

Comment:
Two comments here. First, when you're overriding methods like `viewDidLoad` or `viewDidAppear:`, you should always call the `super` implementation first:
```objc
[super viewDidAppear:animated];
//...
```
That way, you'll run the code in the superclass's implementation of this method, and then your code.

Second, you probably noticed that after you select an option in the detail view and then go back, there's a delay before any changes are reflected in the master table view. If you wanted the change to be more instantaneous, you could create a `delegate` property on your detail view controller and set it to `self` when preparing to segue from the master to the detail controller, and send a message to the master controller whenever an option was selected indicating that it should reload its table view.